hnn_core/tests/test_network.py

```diff
@@ -124,6 +184,8 @@ def test_network():
         assert sorted(this_src_gids) == list(net.gid_ranges[drive['name']])
         # Check type-specific dynamics and events
         n_drive_cells = drive['n_drive_cells']
+        if n_drive_cells_list[drive_idx] != 'n_cells':
+            assert n_drive_cells_list[drive_idx] == n_drive_cells
```

Comment:
I'm now beginning to wonder if adding drives explicitly is worth it. It adds a lot more lines of code to this test without allowing us to directly compare the `n_drive_cells` to the number of instantiated drive cells. Note that when `n_drive_cells = 'n_cells'` (as is the default case for evoked drives), the actual number of drive cells is set implicitly.

Reply:
why don't you add this test right after `net.add_xxx_drive` and check against `net.n_cells` ?

Reply:
That will only work for legacy mode, which I was hoping to avoid adding another dependency on.

Reply:
Fixed! Note that I've now removed dependency on legacy mode here and a few other places in `test_network.py`.